src/thor/timedep_forward.cc

```diff
@@ -136,57 +145,114 @@ bool TimeDepForward::ExpandForward(GraphReader& graphreader,
+    const uint32_t pred_idx,
+    const DirectedEdge* opp_pred_edge,
+    const TimeInfo& time_info,
+    const valhalla::Location& destination,
+    std::pair<int32_t, float>& best_path);
+
 // Runs in the inner loop of `ExpandForward`, essentially evaluating if
 // the edge described in `meta` should be placed on the stack
 // as well as doing just that.
 //
 // Returns true if any edge _could_ have been expanded after restrictions etc.
-inline bool TimeDepForward::ExpandForwardInner(GraphReader& graphreader,
-                                               const EdgeLabel& pred,
-                                               const NodeInfo* nodeinfo,
-                                               const uint32_t pred_idx,
-                                               const EdgeMetadata& meta,
-                                               const graph_tile_ptr& tile,
-                                               const TimeInfo& time_info,
-                                               const valhalla::Location& destination,
-                                               std::pair<int32_t, float>& best_path) {
+template <TimeDepForward::ExpansionType expansion_direction, typename EdgeLabelT>
+inline bool TimeDepForward::ExpandInner(GraphReader& graphreader,
+                                        const EdgeLabelT& pred,
+                                        const baldr::DirectedEdge* opp_pred_edge,
+                                        const NodeInfo* nodeinfo,
+                                        const uint32_t pred_idx,
+                                        const EdgeMetadata& meta,
+                                        const graph_tile_ptr& tile,
+                                        const TimeInfo& time_info,
+                                        const valhalla::Location& destination,
+                                        std::pair<int32_t, float>& best_path) {
+  constexpr bool FORWARD = expansion_direction == TimeDepForward::ExpansionType::forward;
+
+  if (!FORWARD) {
+    // Skip shortcut edges for time dependent routes. Also skip this edge if permanently labeled (best
+    // path already found to this directed edge) or if no access for this mode.
+    if (meta.edge->is_shortcut() || !(meta.edge->reverseaccess() & access_mode_)) {
+      return false;
+    }
+  }
 
   // Skip this edge if permanently labeled (best path already found to this
   // directed edge)
   if (meta.edge_status->set() == EdgeSet::kPermanent) {
     return true; // This is an edge we _could_ have expanded, so return true
   }
+
+  graph_tile_ptr t2 = nullptr;
+  GraphId opp_edge_id;
+  const DirectedEdge* opp_edge = nullptr;
+  if (!FORWARD) {
+    // Get end node tile, opposing edge Id, and opposing directed edge.
+    t2 = meta.edge->leaves_tile() ? graphreader.GetGraphTile(meta.edge->endnode()) : tile;
+    if (t2 == nullptr) {
+      return false;
+    }
+    opp_edge_id = t2->GetOpposingEdgeId(meta.edge);
```

Comment:
Note: I renamed these from TimeDepReverse, they were `oppedge` and `opp_edge` there, which is a bit of a dubious naming scheme.

Reply:
yea, opp_edge_id is a better choice!